packages/robo/src/core/config.ts

```diff
@@ -200,14 +201,33 @@ async function readConfig<T = unknown>(configPath: string): Promise<T> {
 			const rawData = fs.readFileSync(configPath, 'utf8')
 			const pluginConfig = JSON.parse(rawData)
 			return pluginConfig ?? {}
+		} else if (configPath.endsWith('.ts')) {
+			logger.debug('Found typescript config')
+			const outputDir = configPath.split('/').slice(0, -1).join('/')
+
+			await tsup.build({
```

Comment:
This will not be possible. `tsup` is a dev dependency used to build Robo.js itself and is not available to Robo projects.

Use the Robo Compiler instead:
https://github.com/Wave-Play/robo.js/blob/main/packages/robo/src/cli/utils/compiler.ts

It accepts a `files` array similar to tsup's `entry` field. Please set the `distDir` to `.robo/config` to keep operations like these internal, even if they're cleaned up afterwards.

Reply:
https://github.com/Wave-Play/robo.js/pull/351#issuecomment-2453383304